### PR TITLE
Introduce validator for updatable_project_fields validation

### PR DESCRIPTION
**Pull request by DongjinS**
Introduce go-playground/validator for updatable_project_fields validation

In this commit, the validator is only applied for name and one
 regular expression is used because go-playground/validator does not provide
perfect tag for project's name.

In addition, AuthWebhookURL of updatable_project_fields also need to apply validator.

- go-playground/validator
  > https://github.com/go-playground/validator
- regular expression for name
  > https://datatracker.ietf.org/doc/html/rfc3986#section-2.3
  > https://gist.github.com/dpk/4757681

**What this PR does / why we need it**:
- what
   - Introduce go-playground/validator for updatable_project_fields validation
- why
   - Constraints for valid updatable_project_fields(include name) was needed
   - go-playground/validator provide proper validation
**Which issue(s) this PR fixes**:

Fixes #338 
- This PR is extended from https://github.com/yorkie-team/yorkie/pull/338

**Special notes for your reviewer**:
I think there could be more reserved name. 
Currently, there are just two reserved names: "new" and "default".

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

**Review by hackerwins — COMMENTED** [review]
Thank you for suggesting a library that is easy to express constraints.

Let's find a way to use `go-playground/validator` well.

File: api/types/updatable_project_fields.go [hunk 1]
```diff
@@ -54,5 +75,29 @@ func (i *UpdatableProjectFields) Validate() error {
 			}
 		}
 	}
+
+	v := validator.New()
```
Comment (hackerwins) [inline api/types/updatable_project_fields.go:79]: `Validate` is designed to be thread-safe and used as a singleton. So I think creating an instance every time Validate is called is not fit the design intent.

https://github.com/go-playground/validator/blob/9e2ea4038020b5c7e3802a21cfa4e3afcfdcd276/validator_instance.go#L94-L99

Comment (DongjinS) [reply api/types/updatable_project_fields.go]: Thanks for review.
Now I make new file `validator.go` and use `Validate` as a package global variable.
- https://github.com/yorkie-team/yorkie/pull/345/files#diff-ad4ef0932407ad66ec522d5674a6995b785ac92d716d6c9ecd146ff870b181e5:~:text=var%20(-,defaultValidator,-%3D%20initValidator()

Please check it and let me know if you have anything in mind.

**Review by hackerwins — APPROVED** [review]
Overall it looks good. Thanks for your contribution.

**Pull request merged**